app/services/place_proxy_order.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class PlaceProxyOrder
+  def initialize(proxy_order, summarizer, logger, stock_changes_loader)
+    @proxy_order = proxy_order
+    @subscription = proxy_order.subscription
+    @summarizer = summarizer
+    @logger = logger
+    @stock_changes_loader = stock_changes_loader
+  end
+
+  def call
+    return unless initialise_order
+
+    mark_as_processed
+
+    summarizer.record_order(order)
+    return summarizer.record_issue(:complete, order) if order.completed?
+
+    load_changes
+    return handle_empty_order if empty_order?
+
+    move_to_completion
+    send_placement_email
+  rescue StandardError => e
+    summarizer.record_and_log_error(:processing, e.message)
```

Comment:
There's a bug here (Filipe found it). The arguments for this method call are not correct. It needs to be:
```diff
--  summarizer.record_and_log_error(:processing, e.message)
++  summarizer.record_and_log_error(:processing, order, e.message)
```
If an error is encountered during processing, this line throws a fatal error.

Reply:
Bugsnag: https://app.bugsnag.com/open-food-france/coopcircuits/errors/60ca435ef11ca90007833c79?event_id=60ca435e007ed9d831e40000&i=sk&m=nw

Reply:
Cool! thanks guys. That means that line is not covered by tests.

Reply:
This is ready again after https://github.com/openfoodfoundation/openfoodnetwork/pull/7281/commits/1ff75ab3282c18cc91b1c0ba31820d6a1b97f5d3. I put great effort at covering this at unit and integration level.